jiant/tasks/tasks.py

```diff
@@ -288,10 +288,14 @@ def get_metrics(self, reset: bool = False) -> Dict:
     def get_scorers(self):
         return self.scorers
 
-    def update_metrics(self, logits, labels, tagmask=None):
-        assert len(self.get_scorers()) > 0, "Please specify a score metric"
-        for scorer in self.get_scorers():
-            scorer(logits, labels)
+    def update_metrics(self, out, batch):
+        raise NotImplementedError
+
+    def handle_preds(self, preds, batch):
```

Comment:
Should `handle_preds` be a public method of the generic `Task` class? I think this may be better as a private helper called during `update_metrics` in `SocialIQA` (which I think is the only task using this functionality).

Reply:
Actually, please ignore this comment for now. I'm looking more closely at this.

Reply:
Actually I think I'm going to move it to models.py since that's where "preds" is calculated, and this is only a problem for SpanPrediction class.